libcontainer/cgroups/fs/devices.go

```diff
@@ -26,7 +26,11 @@ func (s *DevicesGroup) Apply(d *cgroupData) error {
 }
 
 func (s *DevicesGroup) Set(path string, cgroup *configs.Cgroup) error {
-	if system.RunningInUserNS() {
+	pids, err := cgroups.GetPids(path)
+	if err != nil {
+		return err
+	}
+	if system.RunningInUserNS(pids[0]) {
```

Comment:
@rajasec Yeah, that's what I meant -- I'd prefer that we change the calling interface for `Set` to either use `cgroupData` or maybe we could make a new struct like so:

```
type CgroupParameters struct {
    InUserNs bool
    Cgroup *configs.Cgroup
}
```